vault/logical_system_namespaces.go

```diff
@@ -231,7 +231,7 @@ func (b *SystemBackend) handleNamespacesSet() framework.OperationFunc {
 		path := namespace.Canonicalize(data.Get("path").(string))
 
 		if len(path) > 0 && strings.Contains(path[:len(path)-1], "/") {
-			return nil, errors.New("path must not contain /")
+			return handleError(logical.CodedError(http.StatusBadRequest, "path must not contain /"))
```

Comment:
```suggestion
			return return logical.ErrorResponse("path must not contain /"), logical.ErrInvalidRequest
```

I think should also work fwiw.

Reply:
@cipherboy fixed